src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -60,6 +61,24 @@ def cumulative_node_complexity(self) -> Dict[PlanNodeCategory, int]:
     def cumulative_node_complexity(self, value: Dict[PlanNodeCategory, int]):
         self._cumulative_node_complexity = value
 
+    def replace_child(self, old_node, new_node) -> None:
```

Comment:
you will need to update other nodes like limit to make sure the other fields that is associated with child is updated correctly

Reply:
ohh right. this is going to be a much bigger update.